tests/estimators/speech_recognition/test_tensorflow_lingvo.py

```diff
@@ -0,0 +1,329 @@
+        assert_allclose(np.abs(gradients[2]).sum(), np.abs(gradients[3]).sum(), rtol=1e-01)
+        assert_array_equal(gradients_abs_sum > 0, [False, True, True, True])
+
+    def test_loss_gradient_per_sequence(self, audio_data):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input = audio_data
+        test_target = np.array(["This", "is", "a dummy", "a dummy"])
+
+        lingvo = LingvoAsr()
+
+        gradients = lingvo._loss_gradient_per_sequence(test_input, test_target)
+        gradients_abs_sum = np.array([np.abs(g).sum() for g in gradients], dtype=object)
+
+        # test shape, equal inputs have equal gradients, non-zero inputs have non-zero gradient sums
+        assert [x.shape for x in test_input] == [g.shape for g in gradients]
+        assert_allclose(np.abs(gradients[2]).sum(), np.abs(gradients[3]).sum(), rtol=1e-01)
+        assert_array_equal(gradients_abs_sum > 0, [False, True, True, True])
+
+
+@pytest.mark.skipif(tf.__version__ != "2.1.0", reason="requires Tensorflow 2.1.0")
+@pytest.mark.skipMlFramework("pytorch", "mxnet", "kerastf", "non_dl_frameworks")
+class TestLingvoAsrLibriSpeechSamples:
+    # specify LibriSpeech samples for download and with transcriptions
+    samples = {
+        "3575-170457-0013.wav": {
+            "uri": (
+                "https://github.com/tensorflow/cleverhans/blob/6ef939059172901db582c7702eb803b7171e3db5/"
+                "examples/adversarial_asr/LibriSpeech/test-clean/3575/170457/3575-170457-0013.wav?raw=true"
+            ),
+            "transcript": (
+                "the more she is engaged in her proper duties the last leisure will she have for it even as"
+                " an accomplishment and a recreation"
+            ),
+        },
+        "5105-28241-0006.wav": {
+            "uri": (
+                "https://github.com/tensorflow/cleverhans/blob/6ef939059172901db582c7702eb803b7171e3db5/"
+                "examples/adversarial_asr/LibriSpeech/test-clean/5105/28241/5105-28241-0006.wav?raw=true"
+            ),
+            "transcript": (
+                "the log and the compass therefore were able to be called upon to do the work of the sextant which"
+                " had become utterly useless"
+            ),
+        },
+        "2300-131720-0015.wav": {
+            "uri": (
+                "https://github.com/tensorflow/cleverhans/blob/6ef939059172901db582c7702eb803b7171e3db5/"
+                "examples/adversarial_asr/LibriSpeech/test-clean/2300/131720/2300-131720-0015.wav?raw=true"
+            ),
+            "transcript": (
+                "he obtained the desired speed and load with a friction brake also regulator of speed but waited for an"
+                " indicator to verify it"
+            ),
+        },
+    }
+
+    def test_predict(self):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        transcripts = list()
```

Comment:
similarly to a previous comment about duplicated code, in this case this is a clear nice use case of using option 2 and creating a `back_end_method` with lines from 282 to 292 in it. This method can then be called by both `test_predict` and `test_loss_gradient` tests (see section 3.3.3 in [wiki doc](https://github.com/Trusted-AI/adversarial-robustness-toolbox/wiki/ART-Unit-Testing))

Reply:
See comment related to the `samples` dictionary.